runtime/lua/vim/fcnotify.lua

```diff
@@ -121,6 +121,17 @@ function Watcher:stop()
   self.handle:close()
 end
 
+--- Debounces the watcher. Completely closes the handle and starts again with a new
+--- handle. Necessary for cases when the file being monitored is edited via editors
+--- like vim, which, depending on the value of `backupcopy`, move the original file
+--- for backing it up.
+---
+--@param self:(required, table) The watcher object which should be debounced.
+function Watcher:debounce()
```

Comment:
Do we have to do this if `backupcopy` is not set?

Reply:
This doesn't depend upon current editing session. It depends on what was used to edit the file while we were watching it. If, while we are watching the file, the file is edited by another session, the options of that session would determine the need for this.

Reply:
OK, I think that makes sense to me.